driver-sync/src/main/com/mongodb/client/http/FindIterator.java

```diff
@@ -0,0 +1,308 @@
+    public FindIterable<TResult> hint(final Bson hint) {
+        return null;
+    }
+
+    @Override
+    public FindIterable<TResult> hintString(final String hint) {
+        return null;
+    }
+
+    @Override
+    public FindIterable<TResult> max(final Bson max) {
+        return null;
+    }
+
+    @Override
+    public FindIterable<TResult> min(final Bson min) {
+        return null;
+    }
+
+    @Override
+    public FindIterable<TResult> maxScan(final long maxScan) {
+        return null;
+    }
+
+    @Override
+    public FindIterable<TResult> returnKey(final boolean returnKey) {
+        return null;
+    }
+
+    @Override
+    public FindIterable<TResult> showRecordId(final boolean showRecordId) {
+        return null;
+    }
+    /*
+     *  snapshot
+     */
+    @Override
+    public FindIterable<TResult> snapshot(final boolean snapshot) {
+        return null;
+    }
+
+    public static String getResult(final String mainQuery, final String hostURL) throws IOException {
+        URL url = new URL(hostURL);
+        HttpURLConnection httpConn = (HttpURLConnection) url.openConnection();
+        httpConn.setRequestMethod("POST");
+
+        httpConn.setRequestProperty("x-requested-with", "XMLHttpRequest");
+        httpConn.setRequestProperty("content-type", "application/x-www-form-urlencoded");
+
+        httpConn.setDoOutput(true);
+        OutputStreamWriter writer = new OutputStreamWriter(httpConn.getOutputStream());
+        writer.write(mainQuery);
+        writer.flush();
+        writer.close();
+        httpConn.getOutputStream().close();
+
+        InputStream responseStream = httpConn.getResponseCode() / 100 == 2
+                ? httpConn.getInputStream()
+                : httpConn.getErrorStream();
+        Scanner s = new Scanner(responseStream).useDelimiter("\\A");
+        String response = s.hasNext() ? s.next() : "";
+        return response;
+    }
+
+
+    private Query getQuery(final String filter, final String sortDirection) {
+        Query query = new Query();
+        query.appendQueryKeyValue("Service", "MONGO");
+        query.appendQueryKeyValue("partnerId", "0");
+        query.appendQueryKeyValue("serverType", "GLOBAL");
+        query.appendQueryKeyValue("collectionName", collectionName);
+        query.appendQueryKeyValue("queryType", "general");
+        query.appendQueryKeyValue("limit", String.valueOf(this.limit));
+        query.appendQueryKeyValue("sortField", this.sortField);
+        query.appendQueryKeyValue("includeFields", this.projection);
+        query.appendQueryKeyValue("query", filter);
+        return query;
+    }
+
+    private String getResponse(final String filter, final String sortField, final String sortDirection){
+        Query query = getQuery(filter, sortField);
+        String queryString = query.toString();
+        String response = "";
+        try {
+            response = getResult(queryString, hostURL);
+        } catch (IOException e) {
+            return "";
```

Comment:
exceptions should never be suppressed, this will create hard to debug issues, as every time I/O breaks we will get empty results. For example think of a flow where if there are no big transactions, we allow the transaction, with this, we will do the wrong operation

Reply:
Ya now I understand what problems it can create, replaced that with the following snippet.
```
 String response = "";
        try {
            response = getResult(queryString, hostURL);
        } catch (IOException e) {
            e.printStackTrace();
            throw new RuntimeException(e);
        }
        return response;
    }
```